### ISSUE TITLE
Required workflow is not triggered on `edit`

### ISSUE THREAD

**Issue by yermulnik** [MorrisonCole/pr-lint-action#890]
I'm using this `pr-lint-action` with `on-failed-regex-fail-action: true` and `edited` in the list of `pull_request.types` as a GH Required Workflow and I've found that when PR author edits PR title to conform with the `title-regex`, the workflow doesn't trigger and hence the PR is stuck unless new commits are pushed in to the PR.

Is this expected or am I missing something?

---

![image](https://github.com/user-attachments/assets/58f28dce-e314-4735-80b6-0637922f84d0)

GH Action config:
```yaml
on: # yamllint disable-line rule:truthy
  pull_request:
    branches:
      - main
      - master
    types:
      - edited
      - opened
      - reopened
      - synchronize
  workflow_call:
```
